tools/validate-xep0001-conformance.sh

```diff
@@ -0,0 +1,205 @@
+    validation_result=1
+  else
+    echo "[PASS] The filename ('$file_name') does not start with 'xep-[0-9]'."
+  fi
+fi
+
+# 4. Check that /xep/header/status/text() (XPath) is a defined status (see tools/xeplib.py for an enum)
+case $header_status in
+  ProtoXEP | Experimental | Proposed | Draft | Active | Final | Retracted | Obsolete | Deferred | Rejected | Deprecated )
+    echo "[PASS] XPATH value /xep/header/status/text() ('$header_status') equals a defined status."
+    ;;
+  *)
+    echo "[FAIL] XPATH value /xep/header/status/text() ('$header_status') does not equals a defined status (but should)."
+    validation_result=1
+    ;;
+esac
+
+# 5. Check that /xep/header/type/text() (XPath) is defined in XEP-0001
+#    If the XEP number is less than 400, also accept some legacy values. To find which, see which you encounter in the XEP numbers below 400 :-).
+
+# FIXME: lots of duplications here. Find a better solution.
+case $file_name in
+  'xep-0001.xml' | 'xep-0002.xml' | 'xep-0019.xml' | 'xep-0053.xml' | 'xep-0143.xml' | 'xep-0182.xml' | 'xep-0345.xml' | 'xep-0381.xml' | 'xep-0429.xml' | 'xep-0458.xml' )
+    case $header_type in
+      Historical | Humorous | Informational | Organizational | 'Standards Track' | 'Procedural' )
+        echo "[PASS] XPATH value /xep/header/type/text() ('$header_type') equals a defined type."
+        ;;
+      *)
+        echo "[FAIL] XPATH value /xep/header/type/text() ('$header_type') does not equals a defined type (but should)."
+        validation_result=1
+        ;;
+    esac
+    ;;
+  'xep-0006.xml' | 'xep-0010.xml' | 'xep-0069.xml' | 'xep-0139.xml' )
+    case $header_type in
+      Historical | Humorous | Informational | Organizational | 'Standards Track' | 'SIG Formation' )
+        echo "[PASS] XPATH value /xep/header/type/text() ('$header_type') equals a defined type."
+        ;;
+      *)
+        echo "[FAIL] XPATH value /xep/header/type/text() ('$header_type') does not equals a defined type (but should)."
+        validation_result=1
+        ;;
+    esac
+    ;;
+  'xep-0007.xml' )
+    case $header_type in
+      Historical | Humorous | Informational | Organizational | 'Standards Track' | 'SIG Proposal' )
+        echo "[PASS] XPATH value /xep/header/type/text() ('$header_type') equals a defined type."
+        ;;
+      *)
+        echo "[FAIL] XPATH value /xep/header/type/text() ('$header_type') does not equals a defined type (but should)."
+        validation_result=1
+        ;;
+    esac
+    ;;
+  * )
+    case $header_type in
+      Historical | Humorous | Informational | Organizational | 'Standards Track' )
+        echo "[PASS] XPATH value /xep/header/type/text() ('$header_type') equals a defined type."
+        ;;
+      *)
+        echo "[FAIL] XPATH value /xep/header/type/text() ('$header_type') does not equals a defined type (but should)."
+        validation_result=1
+        ;;
+    esac
+    ;;
+esac
```

Comment:
I don't quite get what this case is about. If you're special casing individual old documents (why?), why not check for a specific type?

When I wrote that requirement, I meant that it would probably be ok to degrade this check to a warning instead of a hard error for XEPs with numbers < 400.

Reply:
My reasoning: the script validates types against a set of valid values. For some specific documents, the set of valid values is a bit larger.